### PR TITLE
Fix #23 -- implement output message template format

### LINKED ISSUE #23: Add verbosity levels
In my usecase, I am defining rule with an extensive hint. While it might make sense to show it in majority of the cases, maybe sometimes in a known set of rules I just need to know which rule failed without the detailed description.
I could imagine it this way:

```
relint myfile.py

1>    myfile.py:12 relint rule name
2>    myfile.py:17 relint second rule name


~~~

relint myfile.py --verbose

1>    myfile.py:12 relint rule name
Hint: extensive rule description for multiple lines

2>    myfile.py:17 relint second rule name
Hint: another extensive rule description for multiple lines
```

### PR DESCRIPTION
**Pull request by amureki**
Inspired by pylint:
https://pylint.pycqa.org/en/latest/user_guide/usage/output.html#custom-message-formats

## Todo if accepted:
- [x] tests
- [x] documentation of all template options

### CODE REVIEW

**Comment by codingjoe** [comment]
@amureki would you care to review this? I'd like to release this in the next major version (v2).

**Comment by codingjoe** [comment]
@amureki almost 😉

**Comment by amureki** [comment]
> @amureki almost 😉

I'll finish it later today, haha 😁

**Comment by amureki** [comment]
@codingjoe not sure I follow the test rule here:
https://github.com/codingjoe/relint/blob/638f53465a0aeea034e2a51ec60e7dd9c87e724a/tests/fixtures/.relint.yml#L7-L10

it mentions unicode, but rule itself copies pattern from "No ToDo" rule above...

**Comment by codingjoe** [comment]
> @codingjoe not sure I follow the test rule here:
> https://github.com/codingjoe/relint/blob/638f53465a0aeea034e2a51ec60e7dd9c87e724a/tests/fixtures/.relint.yml#L7-L10
> 
> it mentions unicode, but rule itself copies pattern from "No ToDo" rule above...

My bad, you can just remove the pattern. It's not needed.

File: relint/__main__.py [hunk 1]
```diff
@@ -42,6 +42,14 @@ def parse_args(args):
     parser.add_argument(
         "-W", "--fail-warnings", action="store_true", help="Fail for warnings."
     )
+    parser.add_argument(
+        "--msg-template",
+        metavar="MSG_TEMPLATE",
+        type=str,
+        default="{filename}:{line_no} {test.name}\nHint: {test.hint}\n{match}",
+        help="Template used to display messages. "
+        "Default: {filename}:{line_no} {test.name}\nHint: {test.hint}\n{match}",
```
Comment (codingjoe) [inline relint/__main__.py:51]: `argparse` remove newlines, maybe display the `\n` in the help text.

```suggestion
        r"Default: {filename}:{line_no} {test.name}\nHint: {test.hint}\n{match}",
```

File: tests/test_main.py [hunk 2]
```diff
@@ -28,6 +28,23 @@ def test_main_execution(self, filename, tmpdir, fixture_dir):
...
 
+    def test_main_execution_with_custom_template(self, capsys, tmpdir, fixture_dir):
+        with (fixture_dir / ".relint.yml").open() as fs:
+            config = fs.read()
+        tmpdir.join(".relint.yml").write(config)
+        tmpdir.join("dummy.py").write("# TODO do something")
+
+        with tmpdir.as_cwd():
+            with pytest.raises(SystemExit) as exc_info:
+                template = "😵{filename}:{line_no} | {test.name} \n {match}"
```
Comment (codingjoe) [inline tests/test_main.py:39]: You are currently not testing the newline replacement bit. This make sure it's actually being used.

```suggestion
                template = r"😵{filename}:{line_no} | {test.name} \n {match}"
```

File: README.md [hunk 3]
```diff
@@ -54,6 +54,28 @@ option `--diff [-d]` or `--git-diff [-g]`:
 git diff --unified=0 | relint my_file.py --diff
 ```
 
+### custom message format
+
+You can customize the output message format by using the `--msg-template=<format string>` option.
```
Comment (codingjoe) [inline README.md:59]:
```suggestion
Customize the output message format with the `--msg-template=<format string>` option.
```

File: README.md [hunk 4]
```diff
@@ -54,6 +54,28 @@ option `--diff [-d]` or `--git-diff [-g]`:
...
+The format string uses the [Python format syntax](https://docs.python.org/3/library/string.html#formatstrings) and the following fields are available:
```
Comment (codingjoe) [inline README.md:60]:
```suggestion
[Python format syntax](https://docs.python.org/3/library/string.html#formatstrings)
is suported for the message template and the following fields are available:
```

File: README.md [hunk 5]
```diff
@@ -54,6 +54,28 @@ option `--diff [-d]` or `--git-diff [-g]`:
 git diff --unified=0 | relint my_file.py --diff
 ```
 
+### custom message format
```
Comment (codingjoe) [inline README.md:57]:
```suggestion
### Custom message format
```

**Review by codingjoe — APPROVED** [review]
🎉

**Pull request merged**